File: validPalindrome/validPalindrome.js
// Time: O(n)
// Space: O(n)
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        const alphanumeric = 'abcdefghijklmnopqrstuvwxyz0123456789';
        // filter string so there are only alphanumeric characters
        s = s.toLowerCase().split('').filter(char => alphanumeric.includes(char)).join('');
        // reverse string
        const reversed = s.split('').reverse().join('');
        // compare strings
        return reversed === s;
    };
};

File: validPalindrome/validPalindrome.py
# Time and space: O(n)
class Solution:
    def isPalindrome(self, s: str) -> bool:
        newStr = ""
        # filter non-alphanumeric characters
        for char in s:
            # if alphanumeric
            if char.isalnum():
                # add lowercase char to new string
                newStr += char.lower()
        # reverse string
        reversed = newStr[::-1]
        # compare
        return newStr == reversed